### PR TITLE
Drop support for Python 3.8

### PR DESCRIPTION
**Pull request by mrtmm**

### CODE REVIEW

**Comment by fghaas** [comment]
@mrtmm Just checking if you want my review here. (I'm marked as reviewer but the PR is still a Draft. :) )

**Comment by mrtmm** [comment]
@fghaas yes I initially added you to review before I saw the tests failing. Now is ready to be reviewed (although still I need to look into the coverage).

**Comment by fghaas** [comment]
Honestly, I'm not sure exactly *what* codecov is complaining about, here. Having a low coverage rate on a diff that's mostly code removal doesn't really make sense to me.

File: Changelog.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+Unreleased
+-------------------------
+* [Chore] Drop support for Python 3.8 and `XBlock<2`.
```
Comment (fghaas) [inline Changelog.md:3]: I don't know exactly how to categorise this but I think it's more than a chore. Maybe just go with this:

```suggestion
* Drop support for Python 3.8 and `XBlock<2` (and, as a consequence, any Open edX releases prior to Redwood).
```

**Review by fghaas — COMMENTED** [review]
This looks good to me in general. I have a couple of tiny nits, and then we also need an update to the README please.

File: tests/resources/course/hastexo/fake_lab_hook_events_1.xml [hunk 2]
```diff
@@ -1,3 +1,4 @@
 <hastexo>
+  <!--Fake comment 1-->
```
[the remark this answers is not in the document]
Comment (mrtmm) [reply tests/resources/course/hastexo/fake_lab_hook_events_1.xml:2]: I _added_ this line to improve test coverage as I saw untested lines about course import with comments.

[the remark this answers is not in the document]
Comment (fghaas) [reply tests/resources/course/hastexo/fake_lab_hook_events_1.xml:2]: Urgh! Looks like I got confused from trying to parse codecov output. Ignore me. :)

File: README.md [hunk 3]
```diff
@@ -24,11 +24,11 @@ repository, you must select the appropriate one:
...
 | Maple            | `>=13.2, <14` | `>=6.0, <7.0`  | `stable-6.0`  |
-| Nutmeg           | `>=14.0, <15` | `>=7.0`        | `master`      |
-| Olive            | `>=15.0, <16` | `>=7.5`        | `master`      |
-| Palm             | `>=16.0, <17` | `>=7.5`        | `master`      |
-| Quince           | `>=17.0, <18` | `>=7.9`        | `master`      |
-| Redwood          | `>=18.0, <19` | `>=7.12`       | `master`      |
+| Nutmeg           | `>=14.0, <15` | `>=7.0, <8.0`  | `stable-7.0`  |
+| Olive            | `>=15.0, <16` | `>=7.5, <8.0`  | `stable-7.0`  |
+| Palm             | `>=16.0, <17` | `>=7.5, <8.0`  | `stable-7.0`  |
+| Quince           | `>=17.0, <18` | `>=7.9, <8.0`  | `stable-7.0`  |
```
Comment (fghaas) [inline README.md:30]: `stable-7.0` doesn't work for a branch where the latest minor release is 7.13. :) You probably want to call the branch `stable-7` instead.

**Comment by fghaas** [comment]
Okay. Final thought: do we want to add

```
python_requires='>=3.11',
```

... to `setup.py`? I know we've never enforced a Python version this way, but I think now might be a good opportunity to start.

**Comment by fghaas** [comment]
Now that we're enforcing a minimum of Python 3.11, we'll also need:

```ini
basepython = python3.11
```

in `tox.ini`, for the `flake8`, `pipdeptree`, and `pipdeptree-requirements` testenvs (and _only_ for those). Otherwise, `tox` will fail when run locally on Ubuntu Jammy, [whose default Python is Python 3.10](https://packages.ubuntu.com/jammy/python3) (which means that [the `python3.11` package](https://packages.ubuntu.com/jammy-updates/python3.11) must be separately installed).

Also, can you please check that that doesn't break anything when running `tox` on a Mac?

**Comment by mrtmm** [comment]
I currently run python3.12 tests locally only and rely the rest to be run in here; adding `basepython = python3.11` does break those tests for me because I don't have python 3.11 installed, should be fine once I install it.

**Comment by fghaas** [comment]
@mrtmm If you're happy with what I suggested in https://github.com/mrtmm/hastexo-xblock/pull/10, then please squash the two commits in this PR into one. Then, I think this one is good to merge (unless you have any second thoughts).

**Review by fghaas — APPROVED** [review]
Looks good to me now; please merge at your convenience. Thanks!

**Pull request merged**